pkg/apis/servicecatalog/v1alpha1/types.go

```diff
@@ -57,15 +57,44 @@ type BrokerSpec struct {
 // BrokerAuthInfo is a union type that contains information on one of the authentication methods
 // the the service catalog and brokers may support, according to the OpenServiceBroker API
 // specification (https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md).
-//
-// Note that we currently restrict a single broker to have only one of these fields
-// set on it.
 type BrokerAuthInfo struct {
-	// BasicAuthSecret is a reference to a Secret containing auth information the
-	// catalog should use to authenticate to this Broker using basic auth.
-	BasicAuthSecret *v1.ObjectReference `json:"basicAuthSecret,omitempty"`
+	// Basic provides configuration for basic authentication
+	Basic *BasicAuthConfig `json:"basic,omitempty"`
+	// Bearer provides configuration for bearer token authentication
+	Bearer *BearerTokenAuthConfig `json:"bearer,omitempty"`
```

Comment:
Yeah, I understand that, but since it's opaque, why is Bearer special? If we just make the entire header opaque, you could support things like JWT without making changes to the api.